src/parquet/bloom_filter.h

```diff
@@ -45,25 +45,39 @@ class BloomFilter final {
   /// @param num_bytes The number of bytes to store Bloom filter bitset.
   explicit BloomFilter(uint32_t num_bytes);
 
-  /// Construct the Bloom filter with given bit set, it is used when reconstructing
+  /// Construct the Bloom filter with given bitset, it is used when reconstructing
   /// a Bloom filter from a parquet file. It uses murmur3_x64_128 as its default hash
   /// function and block-based algorithm as default algorithm.
   ///
   /// @param bitset The given bitset to construct Bloom filter.
   /// @param num_bytes  The number of bytes of given bitset.
-  BloomFilter(const uint32_t* bitset, uint32_t num_bytes);
+  BloomFilter(const uint8_t* bitset, uint32_t num_bytes);
 
-  BloomFilter(const BloomFilter& orig) = delete;
-
-  ~BloomFilter() {
-    if (pool_ && bitset_ && is_internal_bitset_) {
-      pool_->Free(reinterpret_cast<uint8_t*>(bitset_), num_bytes_);
-    }
-    external_bitset_ = NULL;
-  }
+  /// Deserialize the Bloom filter from an input stream, it is used when reconstructing
+  /// a Bloom filter from a parquet filter.
+  ///
+  /// @param input_stream The input stream from which to construct the Bloom filter
+  /// @return it returns a Bloom filter pointer point to a new allocated Bloom filter.
+  static BloomFilter* Deserialize(InputStream* input_stream);
 
   /// Get the number of bytes of bitset
-  uint32_t GetBitsetSize() { return num_bytes_; }
+  uint32_t GetBitsetSize() const { return bloom_filter_algorithm_->GetBitsetSize(); }
+
+  // Hash strategy available for Bloom filter.
+  enum class HashStrategy : uint32_t { MURMUR3_X64_128 = 0 };
+
+  // Bloom filter algorithm.
+  enum class Algorithm : uint32_t { BLOCK = 0 };
```

Comment:
It was used in deserialize function. It can be move back to private since I will remove algorithm and hash check in deserialize function and let constructor to check them.